### ISSUE TITLE
`redundantParens` erroneously removes parentheses in closure definition  

### ISSUE THREAD

**Issue by antiraum** [nicklockwood/SwiftFormat#1674]
This bug was introduced in 0.53.6 and is still present in 0.53.7.

```
let processURL: (URL) async throws -> Void = { url in
}
```

get's reformatted to

```
let processURL: URL async throws -> Void = { url in
}
```

Thanks for SwiftFormat and looking into this!

**Comment by noais** [comment]
Same here: 
`let formatTime: (String) -> Time? = supportsUTCDateFormat`

**Comment by tobihagemann** [comment]
I guess everything has already been said, but another reference is needed, it happened here as well:

```
let localImportHandler: (Error?) -> Void = { error in
```

And now my CI build fails with:

```
single argument function types require parentheses
```

**Comment by nicklockwood** [comment]
@antiraum fixed in 0.53.8
